src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,106 @@
+            System.out.println("Error parsing date: " + e.getMessage());
+        }
+        salaryCalculator(startDay, endDay);
+        return formatSalaryReport(dateFrom, dateTo);
+    }
+
+    private void salaryCalculator(LocalDate dateFrom, LocalDate dateTo) {
+        for (int i = 0; i < emploeesSalaryDetails.length; i++) {
+            if (isDateWithinRange(emploeesSalaryDetails[i].getWorkingDay(), dateFrom, dateTo)) {
+                for (int j = 0; j < emploeesSalaries.length; j++) {
+                    if (emploeesSalaries[j].getEmploeesName().equals(
+                            emploeesSalaryDetails[i].getEmploeeName())) {
+                        emploeesSalaries[j].setEarnedSalary(
+                                  emploeesSalaries[j].getEarnedSalary()
+                                + emploeesSalaryDetails[i].getSalaryPerHour()
+                                * emploeesSalaryDetails[i].getHourCount());
+                        break;
+                    }
+                }
+            }
+        }
+    }
+
+    private void initEmploeesSalary(String[] names) {
+        for (int i = 0; i < names.length; i++) {
+            emploeesSalaries[i] = new EmploeesSalary();
+            emploeesSalaries[i].setEmploeesName(names[i]);
+            emploeesSalaries[i].setEarnedSalary(START_SALARY);
+        }
+    }
+
+    private void initEmploeesSalaryDetails(String[] data) {
+        String[] parsedData;
+        for (int i = 0; i < data.length; i++) {
+            emploeesSalaryDetails[i] = new EmploeesSalaryDetails();
+            parsedData = data[i].split(" ");
+            try {
+                emploeesSalaryDetails[i].setWorkingDay(
+                        LocalDate.parse(parsedData[WORKING_DAY_POSITION], DATE_TIME_FORMATTER));
+                emploeesSalaryDetails[i].setEmploeeName(
+                        parsedData[EMPLOEE_NAME_POSITION]);
+                emploeesSalaryDetails[i].setHourCount(
+                        Integer.parseInt(parsedData[HOUR_COUNT_POSITION]));
+                emploeesSalaryDetails[i].setSalaryPerHour(
+                        Integer.parseInt(parsedData[SALARY_PER_HOUR_POSITION]));
+            } catch (DateTimeParseException ex) {
+                throw new DateTimeParseException("Error parsing date: "
+                        + parsedData[WORKING_DAY_POSITION], parsedData[WORKING_DAY_POSITION],
+                        ex.getErrorIndex());
+            } catch (NumberFormatException ex) {
+                throw new NumberFormatException("Error parsing number: " + data[i]);
+            }
+        }
+    }
+
+    private boolean isDateWithinRange(LocalDate dateToCheck, LocalDate dateFrom, LocalDate dateTo) {
+        if ((dateToCheck.isAfter(dateFrom) && dateToCheck.isBefore(dateTo))
+                || (dateToCheck.isEqual(dateFrom) || dateToCheck.isEqual(dateTo))) {
+            return true;
+        } else {
+            return false;
+        }
```

Comment:
instead of 
if boolean return true else return false
just write
return boolean